### PR TITLE
Optimized BitArray.hash(hasher)

### PR DESCRIPTION
**Pull request by akzhan**

### CODE REVIEW

File: src/bit_array.cr [hunk 1]
```diff
@@ -226,6 +226,13 @@ struct BitArray
     Slice.new(@bits.as(Pointer(UInt8)), (@size / 8.0).ceil.to_i)
   end
 
+  # See `Object#hash(hasher)`
+  def hash(hasher)
+    hasher = size.hash(hasher)
+    hasher = to_slice.hash(hasher)
```
Comment (RX14) [inline src/bit_array.cr]: I think you should edit the `NOTE` above in `==` to mention that `hash` would have to change too.

Comment (akzhan) [reply src/bit_array.cr]: Thanks, done.

**Comment by akzhan** [comment]
Just want to note that CircleCI is someway broken now,

```
./bin/crystal build  -o .build/std_spec spec/std_spec.cr
ld: library not found for -lgmp
```

**Comment by RX14** [comment]
Yeah, looks like @ilovezfs broke it because of https://github.com/crystal-lang/crystal/issues/5089.

**Comment by akzhan** [comment]
@RX14 looks we need your approve.

**Pull request merged**